controllers/front/CartController.php

```diff
@@ -533,14 +527,8 @@ protected function processChangeProductInCart()
                  * If the product can't be in the cart in this quantity, we raise an error.
                  * For the purpose of this error message, we must get the real quantity in stock.
                  * No subtracting of quantity in the cart here.
-                 *
-                 * @todo StockAvailable::getQuantityAvailableByProduct does not work for packs
-                 * which depend on quantity of products inside.
                  */
-                $availableProductQuantity = StockAvailable::getQuantityAvailableByProduct(
-                    $this->id_product,
-                    $this->id_product_attribute
-                );
+                $availableProductQuantity = Product::getQuantity($this->id_product, $this->id_product_attribute);
```

Comment:
This code is just for purposes of getting stock quantity to the error message. Has no functional meaning. We are not passing a cart or customization, so in case of normal products, it will get us the same number as `StockAvailable::getQuantityAvailableByProduct`.